pkl-gradle/src/main/java/org/pkl/gradle/task/BasePklTask.java

```diff
@@ -199,100 +194,6 @@ protected List<Path> parseModulePath() {
-   * </ul>
-   *
-   * In case the returned value is determined to be a {@link URI}, then this URI is first checked
-   * for whether its scheme is {@code file}, like {@code file:///example/path}. In such case, this
-   * method returns a {@link File} corresponding to the file path in the URI. Otherwise, a {@link
-   * URI} instance is returned.
-   *
-   * @throws InvalidUserDataException In case the input is none of the types described above, or
-   *     when the underlying value cannot be parsed correctly.
-   */
-  protected Object parseModuleNotation(Object notation) {
-    if (notation instanceof URI uri) {
-      if ("file".equals(uri.getScheme())) {
-        return new File(uri.getPath());
-      }
-      return uri;
-    } else if (notation instanceof File) {
-      return notation;
-    } else if (notation instanceof Path path) {
-      try {
-        return path.toFile();
-      } catch (UnsupportedOperationException e) {
-        throw new InvalidUserDataException("Failed to parse Pkl module file path: " + notation, e);
-      }
-    } else if (notation instanceof URL url) {
-      try {
-        return parseModuleNotation(url.toURI());
-      } catch (URISyntaxException e) {
-        throw new InvalidUserDataException("Failed to parse Pkl module URI: " + notation, e);
-      }
-    } else if (notation instanceof CharSequence) {
-      var s = notation.toString();
-      if (IoUtils.isUriLike(s)) {
-        try {
-          return parseModuleNotation(IoUtils.toUri(s));
-        } catch (URISyntaxException e) {
-          throw new InvalidUserDataException("Failed to parse Pkl module URI: " + s, e);
-        }
-      } else {
-        try {
-          return Paths.get(s).toFile();
-        } catch (InvalidPathException | UnsupportedOperationException e) {
-          throw new InvalidUserDataException("Failed to parse Pkl module file path: " + s, e);
-        }
-      }
-    } else if (notation instanceof FileSystemLocation location) {
-      return location.getAsFile();
-    } else {
-      throw new InvalidUserDataException(
-          "Unsupported value of type "
-              + notation.getClass()
-              + " used as a module path: "
-              + notation);
-    }
-  }
-
-  protected URI parseModuleNotationToUri(Object m) {
-    var parsed1 = parseModuleNotation(m);
-    return parsedModuleNotationToUri(parsed1);
-  }
-
-  /**
-   * Converts either a file or a URI to a URI. We convert a file to a URI via the {@link
-   * IoUtils#createUri(String)} because other ways of conversion can make relative paths into
-   * absolute URIs, which may break module loading.
-   */
-  private URI parsedModuleNotationToUri(Object notation) {
-    if (notation instanceof File file) {
-      return IoUtils.createUri(file.getPath());
-    } else if (notation instanceof URI uri) {
-      return uri;
-    }
-    throw new IllegalArgumentException("Invalid parsed module notation: " + notation);
-  }
-
```

Comment:
These methods got moved to `PluginUtils.java`, we are now using them in `PklPlugin.java` too.